### ISSUE TITLE
[FR]: SearxNG bot detection

### ISSUE THREAD

**Issue by Quix0r** [fail2ban/fail2ban#3803]
### Environment:

- Fail2Ban version Latest `master` commit
- OS, including release name/version : Devuan Stable

#### Service, project or product which log or journal should be monitored

- Name of filter or jail in Fail2Ban (if already exists) : searxng-blocks.conf
- Service, project or product name, including release name/version :  SearxNG
- Repository or URL (if known) : @SearxNG (here at github)
- Service type : meta search engine
- Ports and protocols the service is listening :  maybe uwscgi-typical, depends on configuration, also local socket file is possible

#### Log or journal information
`YYYY-MM-DD HH:ii:ss,mmm ERROR:searx.botdetection.ip_limit: BLOCK: too many request from <IP>/<DECIMAL-MASK> in SUSPICIOUS_IP_WINDOW (redirect to /)`

- Log file name(s) : `/var/log/uwscgi/app/searx[ng].log (some may have switched from Searx to SearxNG!)

- Journal identifier or unit name : `none`

#### Any additional information
SearxNG typically runs on UWSCGI, but can also run "stand-alone" and then listens by itself on a local port (`localhost`). `mmm` means milliseconds, not minutes.

### Relevant lines from monitored log files:
`YYYY-MM-DD HH:ii:ss,mmm ERROR:searx.botdetection.ip_limit: BLOCK: too many request from <IP>/<DECIMAL-MASK> in SUSPICIOUS_IP_WINDOW (redirect to /)`

#### failures in sense of fail2ban filter (fail2ban must match):

```
```

#### legitimate messages (fail2ban should not consider as failures):

```
YYYY-MM-DD HH:ii:ss,mmm ERROR:searx.engines.yacy: engine timeout
YYYY-MM-DD HH:ii:ss,mmm ERROR:searx: call to ResultContainer.add_unresponsive_engine after ResultContainer.close
YYYY-MM-DD HH:ii:ss,mmm WARNING:searx.engines.yacy: ErrorContext('searx/search/processors/online.py', 116, "response = req(params['url'], **request_args)", 'httpx.TimeoutException', None, (None, None, None)) False
YYYY-MM-DD HH:ii:ss,mmm ERROR:searx.engines.yacy: HTTP requests timeout (search duration : 6.2068777307868 s, timeout: 6.0 s) : TimeoutException
```

**Comment by sebres** [comment]
Do I understand correctly, the failure line looks like this?
```
2024-07-26 22:30:55,850 ERROR:searx.botdetection.ip_limit: BLOCK: too many request from 192.0.2.123/23456 in SUSPICIOUS_IP_WINDOW (redirect to /)
```
Just to ensure the kind of blacken or else the "ciphony" you used is arrived by me correctly.

In that case you may try this RE:
```ini
failregex = ^\s*ERROR:searx.botdetection.ip_limit: BLOCK: too many requests? from <ADDR>
```

**Comment by Quix0r** [comment]
Yes, you are correct about this one. I currently wait for blocking lines to appear.
